precompile/modules/initia_stdlib/sources/dex.move

```diff
@@ -27,6 +27,20 @@ module initia_std::dex {
         coin_b_store: Object<FungibleStore>
     }
 
+    struct FlashSwap has key {
+        coin_a_borrow_amount: u64,
+        coin_b_borrow_amount: u64
+    }
+
+    /// FlashSwapLoan is following the hot potato pattern, so the borrower
+    /// need to pass the created FlashSwapLoan object to this contract to 
+    /// destruct it.
+    ///
+    /// https://move-book.com/programmability/hot-potato-pattern.html
+    struct FlashSwapLoan {
```

Comment:
maybe just name this `FlashSwapReceipt`. feels like it makes more sense than "Loan"